frontend/src/pages/Login/Login.tsx

```diff
@@ -25,20 +25,23 @@ const Login = () => {
     });
     const [visiblePassword, setIsVisiblePassword] = useState<boolean>(false);
     const navigate = useNavigate();
+    const { login } = userAuthService;
 
     const onSubmit = async (userCredientials: UserCredentials) => {
         try {
-            await submitUserCredientials(userCredientials);
+            await login(userCredientials);
             reset();
             toast.success(MESSAGES.SUCCES.USER_REGISTERED);
-            navigate('/main-page');
+            setTimeout(() => navigate('/main-page'), 2000);
         } catch (error: unknown) {
-            if (error instanceof Error) {
-                toast.error(error.message);
+            if (typeof error === 'object' && error !== null && 'message' in error) {
+                toast.error((error as { message: string }).message);
+            } else if (error instanceof Error) {
```

Comment:
Tutaj pytanie raczej odnośnie samego typescripta bo nie wiem. Czy to jest możliwe, żeby w catch weszło coś co nie jest typem Error? Zastanawiam się czy nie dałoby się tego uprościć i wywalić całkowicie ostatniego _else_, a ten _if else_ zamienić na zwykłego _else_. To samo się tyczy obsługi błędów  w Register.tsx